core/src/main/scala/za/co/absa/spline/harvester/dispatcher/httpdispatcher/HttpLineageDispatcherConfig.scala

```diff
@@ -34,4 +38,13 @@ class HttpLineageDispatcherConfig(config: Configuration) {
   val producerUrl: String = config.getRequiredString(ProducerUrlProperty)
   val connTimeout: Duration = config.getRequiredLong(ConnectionTimeoutMsKey).millis
   val readTimeout: Duration = config.getRequiredLong(ReadTimeoutMsKey).millis
+
+  val configureUsingServer: Boolean = config.getRequiredBoolean(ConfigureUsingServer)
+  def apiVersion: Version = stringToVersion(config.getRequiredString(ApiVersion))
```

Comment:
If we make `apiVersion` and `requestCompression` optional (which they are), we could get rid of the `configureUsingServer` parameter. The logic would be simple - if the value is not provided by the user then try to negotiate one with the server.

Reply:
The goal here was make it more transparent and user friendly. Defining some optional properties to avoid server communication on init seems a bit hacky.